app/src/handlers/census/events/MetagameEventEvent.ts

```diff
@@ -0,0 +1,77 @@
+import { injectable } from 'inversify';
+import { GenericEvent } from '../../../types/censusEventTypes';
+import {MetagameEvent} from 'ps2census/dist/client/utils/PS2Events';
+import IllegalArgumentException from '../../../exceptions/IllegalArgumentException';
+import EventId from '../../../utils/eventId';
+
+export enum MetagameEventState {
+    Started,
+    Ended
+}
+
+@injectable()
+export default class MetagameEventEvent {
+    public readonly worldId: number;
+    public readonly eventState: MetagameEventState;
+    public readonly factionNc: number;
+    public readonly factionTr: number;
+    public readonly factionVs: number;
+    public readonly timestamp: number;
+    public readonly instanceId: number;
+    public readonly zoneId: number;
+
+    public constructor(
+        event: GenericEvent
+    ) {
+        const mge = event as MetagameEvent;
+        this.worldId = MetagameEventEvent.parseArgumentAsNumber(mge.world_id);
+        if (isNaN(this.worldId)) {
+            throw new IllegalArgumentException('IllegalArgument: world_id');
+        }
+        if (null === mge.metagame_event_state_name || mge.metagame_event_state_name === undefined) {
+            throw new IllegalArgumentException('IllegalArgument: metagame_event_state_name');
+        }
+        const eventStateName = mge.metagame_event_state_name;
+        if ('started' !== eventStateName && 'ended' !== eventStateName) {
+            throw new IllegalArgumentException('IllegalArgument: metagame_event_state_name');
+        }
+        this.eventState = 'started' === eventStateName? MetagameEventState.Started: MetagameEventState.Ended;
+        this.factionNc = MetagameEventEvent.parseArgumentAsNumber(mge.faction_nc, true);
+        if (isNaN(this.factionNc)) {
+            throw new IllegalArgumentException('IllegalArgument: faction_nc');
+        }
+        this.factionTr = MetagameEventEvent.parseArgumentAsNumber(mge.faction_tr, true);
+        if (isNaN(this.factionTr)) {
+            throw new IllegalArgumentException('IllegalArgument: faction_tr');
+        }
+        this.factionVs = MetagameEventEvent.parseArgumentAsNumber(mge.faction_vs, true);
+        if (isNaN(this.factionVs)) {
+            throw new IllegalArgumentException('IllegalArgument: faction_vs');
+        }
```

Comment:
Feel like we could do a loop here maybe with some variable variables?

Reply:
If you make the cases equal you can do:
```
 ['factionNc', 'factionTr, 'factionVs'].forEach(faction => {
        this[faction] = MetagameEventEvent.parseArgumentAsNumber(mge[faction], true);
        if (isNaN(this[faction])) {
            throw new IllegalArgumentException(`IllegalArgument: ${faction}`);
        }
});
```
Might change this in the ps2census package to change this to camel case.

Reply:
To be honest I don't like this in a constructor, but if you want me to I am up to change it.

Reply:
Yeah this seems strange to have it in the construct, could proxy to a "build" function?

Reply:
We cannot extract it into a function since it is a readyonly property.